File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional


class TutorRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=5000)
    subject: Optional[str] = None
    context: Optional[list[dict]] = None


class TutorResponse(BaseModel):
    response: str
    suggestions: list[str] = []


class GenerateQuizRequest(BaseModel):
    subject: str
    topic: str
    difficulty: str = "medium"
    num_questions: int = Field(default=5, ge=1, le=20)


class GenerateQuizResponse(BaseModel):
    questions: list[dict]
    title: str

File: frontend/src/services/ai.ts
import { api } from "./api"

export const aiService = {
  tutor: (data: { message: string; subject?: string; context?: { role: string; content: string }[] }) =>
    api.post<{ response: string; suggestions: string[] }>("/ai/tutor", data),

  generateQuiz: (data: { subject: string; topic: string; difficulty: string; num_questions: number }) =>
    api.post<{ title: string; questions: Record<string, unknown>[] }>("/ai/generate-quiz", data),
}
